heron/api/src/scala/com/twitter/heron/streamlet/Config.scala

```diff
@@ -0,0 +1,260 @@
+    def setPerContainerRam(perContainerRam: Long): Builder = {
+      this.ram = perContainerRam
+      this
+    }
+
+    /**
+      * Sets the per-container (per-instance) RAM to be used by this topology as a number of bytes
+      * @param perContainerRam Per-container (per-instance) RAM expressed as a Long.
+      */
+    def setPerContainerRamInBytes(perContainerRam: Long): Builder = {
+      this.ram = perContainerRam
+      this
+    }
+
+    /**
+      * Sets the per-container (per-instance) RAM to be used by this topology in megabytes
+      * @param perContainerRamMB Per-container (per-instance) RAM expressed as a Long.
+      */
+    def setPerContainerRamInMegabytes(perContainerRamMB: Long): Builder = {
+      this.ram = perContainerRamMB * MB
+      this
+    }
+
+    /**
+      * Sets the per-container (per-instance) RAM to be used by this topology in gigabytes
+      * @param perContainerRamGB Per-container (per-instance) RAM expressed as a Long.
+      */
+    def setPerContainerRamInGigabytes(perContainerRamGB: Long): Builder = {
+      this.ram = perContainerRamGB * GB
+      this
+    }
+
+    /**
+      * Sets the number of containers to run this topology
+      * @param numContainers The number of containers across which to distribute this topology
+      */
+    def setNumContainers(numContainers: Int): Builder = {
+      config.setNumStmgrs(numContainers)
+      this
+    }
+
+    /**
+      * Sets the delivery semantics of the topology
+      * @param semantics The delivery semantic to be enforced
+      */
+    def setDeliverySemantics(semantics: DeliverySemantics): Builder = {
+      this.deliverySemantics = semantics
+      config.setTopologyReliabilityMode(Config.translateSemantics(semantics))
+      this
+    }
+
+    /**
+      * Sets some user-defined key/value mapping
+      * @param key The user-defined key
+      * @param value The user-defined value
+      */
+    def setUserConfig(key: String, value: AnyRef): Builder = {
+      config.put(key, value)
+      this
+    }
+
+    private def useKryo(): Unit = {
+      config.setSerializationClassName(classOf[KryoSerializer].getName)
+    }
+
+    /**
+      * Sets the {@link Serializer} to be used by the topology (current options are {@link
+      * KryoSerializer} and the native Java serializer.
+      * @param topologySerializer The data serializer to use for streamlet elements in the topology.
+      */
+    def setSerializer(topologySerializer: Serializer): Builder = {
+      this.serializer = topologySerializer
+      this
+    }
+
+    def build(): Config = {
+      if (serializer == Serializer.KRYO) {
+        useKryo()
+      }
+      new Config(this)
+    }
+
+  }
+
+}
+
+/**
+  * Config is the way users configure the execution of the topology.
+  * Things like streamlet delivery semantics, resources used, as well as
+  * user-defined key/value pairs are passed on to the topology runner via
+  * this class.
+  */
+@SerialVersionUID(6204498077403076352L)
```

Comment:
UIDs seem to be the same as Java version. Just to confirm if this what we want or not.